integration/utils/follower.go

```diff
@@ -128,9 +128,20 @@ Loop:
 		getBlockByHeightDuration := time.Since(GetBlockByHeightTime)
 
 		var blockTxs, blockUnknownTxs uint64
+
+		for _, guaranteed := range block.CollectionGuarantees {
+			_, err := f.client.GetCollection(f.ctx, guaranteed.CollectionID)
+			if err != nil {
+				continue Loop // silently try again if not ready; can happen several dozen times
```

Comment:
we don't need to go all the way back to `Loop` (in particular, GetBlockByHeight's result doesn't change)?  we only need to retry getting this specific collection (and maybe double check for f.ctx.Done()).

Reply:
Thanks for the comment! For my taste the loop is complicated enough already, which is probably why this bug crept in in the first place :-) So I'm not keen on doing the loops within loops as suggested, and potentially introducing further bugs. If anything I was hoping somebody might have a suggestion to simplify the entire loop so that my hacky fix wouldn't be necessary :-)

Reply:
your comment about the complicated loop is fair.  In that case, I would suggest that we refactor the two inner loops into `GetAllCollections(CollectionGuarantees) ([]Collection, error)` and `CheckTransactions(Block, []Collection) error` to break down the complexity